verification/cbmc/proofs/aws_string_compare/aws_string_compare_harness.c

```diff
@@ -12,6 +12,8 @@ void aws_string_compare_harness() {
     struct aws_string *str_a = nondet_bool() ? ensure_string_is_allocated_bounded_length(MAX_STRING_LEN) : NULL;
     struct aws_string *str_b =
         nondet_bool() ? (nondet_bool() ? str_a : NULL) : ensure_string_is_allocated_bounded_length(MAX_STRING_LEN);
+    __CPROVER_assume(IMPLIES(str_a != NULL, aws_string_is_valid(str_a)));
```

Comment:
This `IMPLIES(x != NULL, foo_is_valid(x))` pattern occurs often enough to be worth its own macro

Reply:
I agree, so I created this issue https://github.com/awslabs/aws-c-common/issues/713 to add this on a separate PR.